File: rb101/RB101-RB109: Small Problems/Easy 8/double_char1.rb
# double_char1.rb
# Write a method that takes a string, and returns a new string in which every
# character is doubled.

def repeater(string)
  string.split('').map { |char| char * 2 }.join
end

puts repeater('Hello') == "HHeelllloo"
puts repeater("Good job!") == "GGoooodd  jjoobb!!"
puts repeater('') == ''

puts '-----'

# Launch School solution:

def repeater(string)
  result = ''
  string.each_char do |char|
    result << char << char
  end
  result
end

puts repeater('Hello') == "HHeelllloo"
puts repeater("Good job!") == "GGoooodd  jjoobb!!"
puts repeater('') == ''

File: js210/js210_small_problems/easy_5/double_char_1.js
function repeater(text) {
  let doubled = '';

  for (let index = 0; index < text.length; index += 1) {
    doubled += text[index].repeat(2);
  }

  return doubled;
}

console.log(repeater('Hello'));
console.log(repeater('Good job!'));
console.log(repeater(''));
